src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  // died() {
+  //   if (this.health <= 0) {
+  //     delete Animal.alive.this;
+  //   }
+  // }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
```

Comment:
Since the `name` property is already set in the parent `Animal` class, reassigning it here in the `Herbivore` subclass is unnecessary and should be removed to avoid redundancy.